cmd/func/main.go

```diff
@@ -14,6 +17,11 @@ import (
 var date, vers, hash string
 
 func main() {
+
+	rt := funcHttp.NewRoundTripper()
+	http.DefaultTransport = rt
+	defer rt.Close()
```

Comment:
I am a little bit concerned with overriding the `http` package default transport, because this affects anything in the process using the `http` package's standard functions.  It also is a bit hidden, because code utilizing this feature is relying on something happening "out of band" (not immediately visible in the code making the http requests).  Finally, this is only going to affect code using the CLI library, meaning that anyone using the `func` library will not get this feature, and it is not obvious why.

I love the feature this provides, but I wonder if it's not more clear to instead explicitly define this custom transport on a client at the time it is being used.  In other workds, opt in rather than opt out.  Can't we just create a new http client with a custom RoundTripper from the `funchttp`?

Reply:
/hold

Reply:
> /hold

Too late now I suppoes, the merge-bot beat me to the punch.
I am still interested to hear what others think about my concern above, but I am sure we can refactor it slightly if it ever becomes a problem.

Reply:
@lkingland that's a really great point. You are right in my opinion.

Reply:
I tend to agree as well. I've added https://github.com/knative-sandbox/kn-plugin-func/issues/715 so we don't lose this.